apps/dashboard/app/models/project.rb

```diff
@@ -8,6 +8,29 @@ class Project
   include IconWithUri
   extend JobLogger
 
+  def self.from_directory(dir)
+    # fetch "id" by opening .ondemand/manifest.yml
+    manifest_path = Pathname("#{dir.to_s}/.ondemand/manifest.yml")
+    unless manifest_path.exist?
+      Rails.logger.warn("Imported directory is not a Open OnDemand project")
+      nil
+    end
+
+    contents = File.read(manifest_path)
+    raw_opts = YAML.safe_load(contents)
+    id = raw_opts["id"]
+
+    project = Project.find(id)
+    if project.nil?
+      Project.new({ id: id, directory: dir })
+    else
+      nil
+    end
+  rescue StandardError => e
+    Rails.logger.warn("Cannot import project from dir #{dir} due to error #{e}")
+    false
```

Comment:
Sometimes this returns `false` and sometimes this returns `nil` - can you clean this up to alwasy return nil on the error case.

Also you dno't need this if/esle block, you can just return `Project.find(id)` and let it return `nil` or a `project`.

Reply:
`Project.find(id)` will return project if it exists in own lookup_directory; but while doing `from_directory` the idea is to save a project that doesn't exist in lookup_directory.

So, doing Project.new() is necessary, so later we can save it to lookup_directory.

Let me know if you have different thoughts for `from_directory` function.

Reply:
OK - I think this has gotten me twisted around, I left a comment on the recent updates.